tests/Helper/HeaderHelperTest.php

```diff
@@ -134,37 +150,75 @@ public function testSortedAcceptTypesFromRequest(RequestInterface $request, arra
-            'invalidQuotes2' => ['a="test', null, \InvalidArgumentException::class],
-            'invalidQuotes3' => ['a=test"', null, \InvalidArgumentException::class],
-            'invalidEmptyQuotes' => ['a=""', null, \InvalidArgumentException::class],
-            'invalidEmptyValue' => ['a=b; c=', null, \InvalidArgumentException::class],
+            'simple' => ['a=test;test=test55',null, null, ['a' => 'test', 'test' => 'test55']],
+            'quoted' => ['a="test";b="test2;";d="."',null, null, ['a' => 'test', 'b' => 'test2;', 'd' => '.']],
+            'mixed' => ['a=b;c="apple"',null, null, ['a' => 'b', 'c' => 'apple']],
+            'one' => ['a=test',null, null, ['a' => 'test']],
+            'oneQuoted' => ['a="test"',null, null, ['a' => 'test']],
+            'oneQuotedEmpty' => ['a=""',null, null, ['a' => '']],
+            'oneSingleQuoted' => ["a='a'",null, null, ['a' => "'a'"]],
+            'mixedQuotes' => [
+                'a="tes\'t";test="\'test\'";test2="\\"quoted\\" test"',null, null,
+                ['a' => 'tes\'t', 'test' => '\'test\'', 'test2' => '"quoted" test']
+            ],
+            'slashes' => [
+                'a="\\t\\e\\s\\t";b="te\\\\st";c="\\"\\"',null, null,
+                ['a' => 'test', 'b' => 'te\\st', 'c' => '"\\']
+            ],
+            'specChars' => ['*=test;test=*',null, null, ['*' => 'test', 'test' => '*']],
+            'specChars2' => ['param*1=a;param*2=b',null, null, ['param*1' => 'a', 'param*2' => 'b']],
+            'numbers' => ['a=8888;b="999"',null, null, ['a' => '8888', 'b' => '999']],
+            'invalidQuotes2' => ['a="test',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes3' => ['a=test"',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes4' => ['a=te"st',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyValue' => ['a=b; c=',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyParam' => ['a=b; ;c=d',null, null, null, \InvalidArgumentException::class],
+            'semicolonAtEnd' => ['a=b;',null, null, null, \InvalidArgumentException::class],
+            'comma' => ['a=test,test',null, null, null, \InvalidArgumentException::class],
+
+            # true syntax
+            // 'spaces2' => [' a = b ; c = "d" ',null, null, ['a' => 'b', 'c' => 'd']],
+            'case' => ['A=TEST;TEST=B',true, false, ['a' => 'TEST', 'test' => 'B']],
+            'case2' => ['A=TEST;TEST=B',false, false, ['A' => 'TEST', 'TEST' => 'B']],
+            'case3' => ['A=TEST;TEST=B',false, true, ['A' => 'test', 'TEST' => 'b']],
+            // 'spaces1' => ['a=b; c="d" ',null, null, ['a' => 'b', 'c' => 'd']],
+            'spaces3' => ['a=b c',null, null, null, \InvalidArgumentException::class],
+            'percent' => ['a=%1;b="foo-%32-bar"',null, null, ['a' => '%1', 'b' => 'foo-%32-bar']],
+
+            # Invalid syntax but most browsers take a first parameter
```

Comment:
This is an interesting case, and I would rather support an exception. Invalid use causes quite a few problems, eg. why not accept the first or why not the last.

I think it is likely that the yii framework will mainly receive request by standard browsers. In the case of an API server, this can also force the client (and developer) site to send valid requests.

Reply:
I offered only test cases. On controversial moments, when the syntax is broken, we are free to do as we prefer: to be more strict or to do as most browsers

Reply:
```php
            # Invalid syntax but most browsers take a first parameter
            // 'sameName' => ['a=T1;a="T2"',null, null, ['a' => 'T1']],
            // 'sameNameCase' => ['aa=T1;Aa="T2"',null, null, ['aa' => 'T1']],
```
@samdark first, last or throw?

Reply:
>first, last or throw?

I think most browsers take the first value because they are trying to take the maximum from the Headers until the RFC is violated.
Declaring the second parameter of the same name is the violation, so all that was normal before, is accepted (first declaration).

Reply:
first is the winner.